src/feat/online-feature-test.cc

```diff
@@ -375,6 +375,43 @@ void TestOnlineAppendFeature() {
   }
 }
 
+void TestRecyclingVector() {
+  RecyclingVector full_vec;
+  RecyclingVector shrinking_vec(10);
+  for (int i = 0; i != 100; ++i) {
+    Vector <BaseFloat> data(1);
+    data.Set(i);
+    full_vec.Store(new Vector<BaseFloat>(data));
+    shrinking_vec.Store(new Vector<BaseFloat>(data));
+  }
```

Comment:
Please add KALDI_ASSERTs for `Size()` at this point. The size is computed non-trivially in the end.

Reply:
+1